### DISCUSSION TITLE
How to use CheckBox and RadioButton in Toga？

### DISCUSSION THREAD

**Discussion by wanglifeng95** [beeware/toga#2100] [Q&A] [answered]
It seems that I haven't seen CheckBox and RadioButton in Toga. Are there any widgets with similar functions available?

**Comment by freakboy3742** [comment, marked as answer]
`toga.Switch` will be implemented as a checkbox on some platforms; there's no `toga.Checkbox` because some platforms (most notably mobile platforms) tend to avoid checkboxes.

Unfortunately, there's no radio button widget at present. This is probably the most notable "missing" widget from Toga's collection at present. You can get functionally equivalent behavior using a `toga.Selection` (i.e., select exactly 1 option from a list of options), but obviously that isn't visually the same.

**Reply by wanglifeng95** [reply]
Okay, thank you. For now, I will use Switch and Selection instead of Checkbox and RadioButton.

**Reply by Kuna42** [reply]
@freakboy3742 
An idea:
What if both exists, Switch and Checkbox, and the platforms which do not support Checkbox by native, have a Switch instead.
Sure, it is bad, that you have two different classes which maps to the same widget layout on some platforms, but i think it is better instead of shrinking different kinds of widgets.

**Reply by freakboy3742** [reply]
Thanks for the suggestion; I've opened #2225 to track this idea.

**Comment by wanglifeng95** [comment]
您好，我已收到来信，会尽快给予你回复。
